File: initial/src/main/java/com/example/uploadingfiles/storage/StorageProperties.java
package com.example.uploadingfiles.storage;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

@Component
@ConfigurationProperties("storage")
public class StorageProperties {

	/**
	 * Folder location for storing files
	 */
	private String location = "upload-dir";

	public String getLocation() {
		return location;
	}

	public void setLocation(String location) {
		this.location = location;
	}

}

File: initial-kotlin/src/main/kotlin/com/example/uploadingfiles/storage/StorageProperties.kt
package com.example.uploadingfiles.storage

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

@Component
@ConfigurationProperties("storage")
class StorageProperties {
    /**
     * Folder location for storing files
     */
    var location: String = "upload-dir"
}
